docs(dev-infra): document limitation in ts-circular-deps tool

devversion opened the pull request:
Adds documentation on discovered limitations in the ts-circular-deps
tool, so that we can reference it when needed.

josephperrott reviewed (approved):
LGTM, just one small nit.


Thanks again for such a thorough consideration of this!

devversion committed:
docs(dev-infra): document limitation in ts-circular-deps tool

Adds documentation on discovered limitations in the ts-circular-deps
tool, so that we can reference it when needed.

devversion commented on dev-infra/ts-circular-dependencies/README.md:
```diff
@@ -0,0 +1,82 @@
+### ts-circular-dependencies
+
+This tool accepts a test configuration that declares a set of source files which
```
👍 Done!
